clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java

```diff
@@ -142,8 +142,13 @@
      */
     @Deprecated
     public static final String BLOCK_ON_BUFFER_FULL_CONFIG = "block.on.buffer.full";
-    private static final String BLOCK_ON_BUFFER_FULL_DOC = "When our memory buffer is exhausted we must either stop accepting new records (block) or throw errors. By default " + "this setting is true and we block, however in some scenarios blocking is not desirable and it is better to "
-                                                           + "immediately give an error. Setting this to <code>false</code> will accomplish that: the producer will throw a BufferExhaustedException if a record is sent and the buffer space is full.";
+    private static final String BLOCK_ON_BUFFER_FULL_DOC = "When our memory buffer is exhausted we must either stop accepting new records (block) or throw errors. "
+                                                           + "By default this setting is false and the producer will throw a BufferExhaustedException if a record is sent and the buffer space is full. "
+                                                           + "However in some scenarios getting an error is not desirable and it is better to block. Setting this to <code>true</code> will accomplish that."
+                                                           + "<em>If this property is set to true, parameter <code>metadata.fetch.timeout.ms</code> is not longer honored.</em>"
+                                                           + "<p>"
+                                                           + "This parameter is deprecated and will be removed in a future release. "
+                                                           + "Parameter <code>max.block.ms</code> should be used instead.";
```

Comment:
This does not compile (ie, if I use `MAX_BLOCK_MS_CONFIG`). I am actually a little bit confused because `METADATA_FETCH_TIMEOUT_CONFIG` is actually deprecated but I don't get an error if I use it... Can you explain?

```
:clients:compileJava
/home/mjsax/workspace_kafka/kafka/clients/src/main/java/org/apache/kafka/clients/producer/ProducerConfig.java:151: error: illegal forward reference
                                                           + "Parameter <code>" + MAX_BLOCK_MS_CONFIG + "</code> should be used instead.";
                                                                                  ^
Note: /home/mjsax/workspace_kafka/kafka/clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java uses or overrides a deprecated API.
```